pkg/apis/core/types.go

```diff
@@ -5732,6 +5732,14 @@ type TopologySpreadConstraint struct {
 	// This is a alpha-level feature enabled by the NodeInclusionPolicyInPodTopologySpread feature flag.
 	// +optional
 	NodeTaintsPolicy *NodeInclusionPolicy
+	// MatchLabelKeys is a set of pod label keys to select the pods over which
+	// spreading will be calculated. The keys are used to lookup values from the
+	// incoming pod labels, those key-value labels are ANDed with `LabelSelector`
+	// to select the group of existing pods over which spreading will be calculated
+	// for the incoming pod. Keys that don't exist in the incoming pod labels will
+	// be ignored.
+	// +optional
+	MatchLabelKeys []string
```

Comment:
Done

Reply:
We probably better validate that a key in matchLabelKeys don't exist in labelSelector. It doesn't seem like there is a use case for it, so better block it.

Reply:
I feel that this is too strict and lacks flexibility. Maybe like labelSelector, it will also may be contradicted with the use of the same key in both labelSelector and matchExpressions. But we don't block it? right? 
```go
kind: Pod
apiVersion: v1
metadata:
  name: mypod
  labels:
    foo: a  
spec:
  topologySpreadConstraints:
  - maxSkew: 1
    topologyKey: zone
    whenUnsatisfiable: DoNotSchedule
    labelSelector:
      matchLabels: ---> foo = a
        foo: a
      matchExpressions:   --> foo in {b}
      - key: foo
        operator: In
        values:
        - b
  containers:
  - name: pause
    image: k8s.gcr.io/pause:3.1
```
The keys in matchLabelKeys and labelSelector do not necessarily contradict each other. Maybe just some cases.

Reply:
But lacks flexibility to do what? What is the use case? The example you gave above translates to a selector that matches nothing, which probably isn't what the user intended to do.

Reply:
Yes. You are right. This case is not the user intend to do. 
I added the validation for this.